lib/index.ts

```diff
@@ -726,3 +729,31 @@ function toSnykVersion(v: ModuleVersion): string {
     throw new Error('Unexpected module version format');
   }
 }
+
+function constructPurl(mod: GoModule, importPath: string): string | undefined {
+  let namespace: string | undefined;
+  let name: string | undefined;
+  const version = mod.Version;
+  let subpath: undefined | string;
```

Comment:
```
  let namespace: string | undefined;
  let name: string | undefined;
```
Then
```
let subpath: undefined | string;
```

<img src="https://media0.giphy.com/media/U5V2m54zOwDJZBSSHk/giphy.gif"/>

Reply:
On a serious note - do they really all have to be potentially `undefined`? 😞

Reply:
can refactor a bit here, but yeah they are all optional except for `name`. `string | undefined` is an improvement from [`string | undefined | null`](https://github.com/package-url/packageurl-js/blob/3e0e10616b23ba5c1a9529f31357abd5ee0bfa72/src/package-url.d.ts#L63) imho 😆